packages/synthetics-sdk-broken-links/test/unit/broken_links.spec.ts

```diff
@@ -0,0 +1,187 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import { expect, use } from 'chai';
+import chaiExclude from 'chai-exclude';
+use(chaiExclude);
+
+import {
+  BrokenLinksResultV1_BrokenLinkCheckerOptions,
+  BrokenLinksResultV1_BrokenLinkCheckerOptions_LinkOrder,
+  BrokenLinksResultV1_SyntheticLinkResult,
+  ResponseStatusCode,
+  ResponseStatusCode_StatusClass,
+} from '@google-cloud/synthetics-sdk-api';
+import {
+  runBrokenLinks,
+  BrokenLinkCheckerOptions,
+} from '../../src/broken_links';
+const path = require('path');
+
+describe('runBrokenLinks', async () => {
+  const status_class_2xx: ResponseStatusCode = {
+    status_class: ResponseStatusCode_StatusClass.STATUS_CLASS_2XX,
+  };
+  it('exits early if origin_url is in an incorrect format', async () => {
+    const inputOptions: BrokenLinkCheckerOptions = {
+      origin_url: 'fake-url.html',
+    };
+
+    const syntheticResult = await runBrokenLinks(inputOptions);
+
+    const broken_links_result =
+      syntheticResult.synthetic_broken_links_result_v1;
+    expect(broken_links_result?.origin_link_result?.link_passed).to.be.false;
+    expect(broken_links_result?.followed_link_results?.length).to.equal(0);
+  }).timeout(10000);
+
+  it('returns generic_result with appropriate error information if error thrown', async () => {
+    const inputOptions: BrokenLinkCheckerOptions = {
+      origin_url: 'url-does-not-start-with-http',
+    };
+
+    const syntheticResult = await runBrokenLinks(inputOptions);
+
+    const genericResult = syntheticResult.synthetic_generic_result_v1;
+    expect(genericResult).to.be.exist;
+    expect(genericResult?.ok).to.be.false;
+    expect(genericResult?.generic_error?.error_type).to.equal('Error');
+    expect(genericResult?.generic_error?.error_message).to.equal(
+      'origin_url must be a string that starts with `http`'
+    );
+  }).timeout(10000);
+
+  it('returns generic_result when waitForSelector exceeds deadline', async () => {
```

Comment:
Change from generic error to one that says that the origin url passed === false, with info explaining that the content that we're looking for is not present.